pkg/boot/initrd_test.go

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package boot
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+)
+
+func TestConcatenationSize(t *testing.T) {
+	a := strings.NewReader("yay")
+	b := bytes.NewReader(make([]byte, 777))
+	buf, _ := catInitrds(a, b)
+	size := len(buf.Bytes())
+	if size != 1536 {
+		t.Errorf("want 1536 bytes, got %v", size)
+	}
+}
+
+func TestConcatenationBytes(t *testing.T) {
+	a := strings.NewReader("foo")
+	b := strings.NewReader("bar")
+	buf, _ := catInitrds(a, b)
+	res := buf.Bytes()
```

Comment:
alternative: rather than having catInitrd, use uio.ReadAll to get the buffer and check the size of the buffer. That way you test behavior, not implementation details.